File: go-backend/dto/route.go
package dto

type RouteDto struct {
	Name          string  `json:"name" binding:"required"`
	NodeIds       []int64 `json:"nodeIds" binding:"required"`
	Protocol      string  `json:"protocol"`
	TcpListenAddr string  `json:"tcpListenAddr"`
	UdpListenAddr string  `json:"udpListenAddr"`
	InterfaceName string  `json:"interfaceName"`
}

type RouteUpdateDto struct {
	ID            int64   `json:"id" binding:"required"`
	Name          string  `json:"name" binding:"required"`
	NodeIds       []int64 `json:"nodeIds" binding:"required"`
	Protocol      string  `json:"protocol"`
	TcpListenAddr string  `json:"tcpListenAddr"`
	UdpListenAddr string  `json:"udpListenAddr"`
	InterfaceName string  `json:"interfaceName"`
	Status        *int    `json:"status"`
}

File: nextjs-frontend/lib/api/route.ts
import { post } from './client';

type RoutePayload = {
  name: string;
  nodeIds: number[];
  protocol?: string;
  tcpListenAddr?: string;
  udpListenAddr?: string;
  interfaceName?: string;
  status?: number;
};

type RouteUpdatePayload = RoutePayload & {
  id: number;
};

export const createRoute = (data: RoutePayload) => post('/route/create', data);
export const getRouteList = () => post('/route/list');
export const updateRoute = (data: RouteUpdatePayload) => post('/route/update', data);
export const deleteRoute = (id: number) => post('/route/delete', { id });
